trio/tests/test_util.py

```diff
@@ -58,3 +58,27 @@ def handler(signum, _):
                 async with ul1:
                     pass  # pragma: no cover
     assert "ul1" in str(excinfo.value)
+
+
+async def test_contextmanager_do_not_unchain_non_stopiteration_exceptions():
+    @acontextmanager
+    async def test_issue29692():
+        try:
+            yield
+        except Exception as exc:
+            raise RuntimeError('issue29692:Chained') from exc
+    try:
+        async with test_issue29692():
+            raise ZeroDivisionError
+    except Exception as ex:
+        assert type(ex) is RuntimeError
+        assert ex.args[0] == 'issue29692:Chained'
+        assert isinstance(ex.__cause__, ZeroDivisionError)
```

Comment:
Went ahead and fixed it up as you suggested. I was just copy-pasting from upstream, but I think it's nicer this way.